splice_machine/src/main/java/com/splicemachine/derby/ddl/DDLUtils.java

```diff
@@ -1047,4 +1050,26 @@ public static void preUpdateSystemProcedures(DDLMessage.DDLChange change, DataDi
             throw StandardException.plainWrapException(e);
         }
     }
+
+    public static void preLeaveRestore(DDLMessage.DDLChange change, DataDictionary dd) throws StandardException {
+        if (LOG.isDebugEnabled())
+            SpliceLogUtils.debug(LOG,"preRestore with change=%s",change);
+        TxnView txn = DDLUtils.getLazyTransaction(change.getTxnId());
+        try (SpliceTransactionResourceImpl transactionResource = new SpliceTransactionResourceImpl()) {
+            transactionResource.marshallTransaction(txn);
+            SIDriver.driver().getTxnSupplier().invalidate();
+            SIDriver.driver().getIgnoreTxnSupplier().refresh();
+            dd.getDataDictionaryCache().clearAll();
+            SIDriver.driver().lifecycleManager().leaveRestoreMode();
+            Collection<Context> allContexts = ContextService.getService().getAllContexts(LanguageConnectionContext.CONTEXT_ID);
+            allContexts = ContextService.getService().getAllContexts(LanguageConnectionContext.CONTEXT_ID);
+            for (Context context : allContexts) {
+                ((LanguageConnectionContext) context).leaveRestoreMode();
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
this is probably not needed.

Reply:
It's followed by rethrowing the exception, so if we're guaranteed that the rethrown exception will be logged or properly handled, the printStackTrace may be redundant.  But keeping it means if the printStackTrace is logged, we know it's logged at least once, and it may be logged in a different log from the one where the rethrown exception will be logged, which could be useful.